audio/filter/af_scaletempo2_internals.c

```diff
@@ -99,11 +99,13 @@ static float multi_channel_similarity_measure(
 {
     const float epsilon = 1e-12f;
     float similarity_measure = 0.0f;
+    float total_energy = epsilon;
```

Comment:
Removing that shouldn't make any difference to the result.

The target block doesn't change with the offset, therefore this sum will always be the same for each iteration of the optimal index search, so all calculated similarities are changed by the same factor, thus not changing the location of the maximum.

Reply:
Hmm, that's true. I was thinking of the sliding window here. On the other hand, this relies on the fact that target block is always passed first, and this method currently doesn't say which one the target block is (though I did phrase my commit message that way). I'll remove it.

Reply:
how about changing the parameters to make the intention apparent? I'm thinking of this
```
 static float multi_channel_similarity_measure(
-    const float* dot_prod_a_b,
-    const float* energy_a, const float* energy_b,
+    const float* dot_prod,
+    const float* energy_target, const float* energy_candidate,
     int channels)
 {
     const float epsilon = 1e-12f;
     float similarity_measure = 0.0f;
     for (int n = 0; n < channels; ++n) {
-        similarity_measure += dot_prod_a_b[n] * energy_a[n]
-            / sqrtf(energy_a[n] * energy_b[n] + epsilon);
+        similarity_measure += dot_prod[n] * energy_target[n]
+            / sqrtf(energy_target[n] * energy_candidate[n] + epsilon);
     }
     return similarity_measure;
 }
```

Reply:
Now that the target and candidate aren't treated the same anymore in that function, it does make sense to give them more descriptive names. I'd be fine with those.